src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -180,11 +192,13 @@ AzureOAuthService getOAuthService() {
 
     private String getRootUrl() {
         Jenkins jenkins = Jenkins.getInstance();
-        return StringUtils.stripEnd(jenkins.getRootUrl(), "/");
+        String url = isFromRequest() ? jenkins.getRootUrlFromRequest() : jenkins.getRootUrl();
+        return StringUtils.stripEnd(url, "/");
     }
 
     @DataBoundConstructor
-    public AzureSecurityRealm(String tenant, String clientId, String clientSecret, int cacheDuration) {
+    public AzureSecurityRealm(String tenant, String clientId, String clientSecret, int cacheDuration)
+            throws ExecutionException, IOException, InterruptedException {
```

Comment:
Your modifying the DataBoundConstructor with exceptions, I am not sure these are correct.

Reply:
Removed. It seems the IDE overdid it